lib/matplotlib/colors.py

```diff
@@ -1225,6 +1240,594 @@ def reversed(self, name=None):
         return new_cmap
 
 
+class MultivarColormap(ColormapBase):
+    """
+    Class for holding multiple `~matplotlib.colors.Colormap` for use in a
+    `~matplotlib.cm.VectorMappable` object
+
+    MultivarColormap does not support alpha in the constituent
+    look up tables (ignored).
+    """
+    def __init__(self, name, colormaps, combination_mode):
+        """
+        Parameters
+        ----------
+        name : str
+            The name of the colormap family.
+        colormaps: list or tuple of `~matplotlib.colors.Colormap` objects
+            The individual colormaps that are combined
+        combination_mode: str, 'Add' or 'Sub'
+            Describe how colormaps are combined in sRGB space
+            'Add' -> additive
+            'Sub' -> subtractive
+        """
+        self.name = name
+
+        if not np.iterable(colormaps) or len(colormaps) == 1:
+            raise ValueError("A MultivarColormap must have more than one colormap.")
+        for cmap in colormaps:
+            if not issubclass(type(cmap), Colormap):
+                raise ValueError("colormaps must be a list of objects that subclass"
+                                 " Colormap, not strings or list of strings")
+
+        self.colormaps = colormaps
+        self.combination_mode = combination_mode
+        self.n_variates = len(colormaps)
+        self._rgba_bad = (0.0, 0.0, 0.0, 0.0)  # If bad, don't paint anything.
```

Comment:
Bad is if a value is np.nan or masked.
If there is an impossible color, it is simply clamped to the possible color space.

Reply:
We can use `missing` instead of `bad`, I chose bad because that is the word used in 'Colormap', and I thought it would be good to use the same word for consistency